test/drain_test.go

```diff
@@ -486,3 +486,273 @@ func TestDrainConnDuringReconnect(t *testing.T) {
 		t.Fatalf("Timeout waiting for closed state for connection")
 	}
 }
+
+func TestDrainStatus(t *testing.T) {
+	t.Run("subscribe", func(t *testing.T) {
+		s := RunDefaultServer()
+		defer s.Shutdown()
+
+		nc, err := nats.Connect(nats.DefaultURL)
+		if err != nil {
+			t.Fatalf("Failed to create default connection: %v", err)
+		}
+		defer nc.Close()
+
+		if nc.IsDraining() {
+			t.Fatalf("Expected IsDraining to be false")
+		}
+
+		sub, err := nc.Subscribe("foo", func(_ *nats.Msg) {
+			time.Sleep(10 * time.Millisecond)
+		})
+		if err != nil {
+			t.Fatalf("Error creating subscription; %v", err)
+		}
+		for i := 0; i < 100; i++ {
+			nc.Publish("foo", []byte("hello"))
+		}
+		time.Sleep(100 * time.Millisecond)
+		sub.Drain()
+		ds := sub.DrainStatus()
```

Comment:
maybe later can have some helper build on top like `sub.WaitDrained(ctx) error` that waits for drain to complete or timeout.

Reply:
Yeah, that's a good idea. Not sure I want to add it in this PR though, it's easy enough to implement it yourself and I want to be careful about adding public APIs.

In the future though, that might be useful.